from_mwt_ds/DataScience/notebooks/explorations.ipynb

```diff
@@ -0,0 +1,492 @@
+    "## Visualizers"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def plot_env(examples, ax=None, window=100):\n",
+    "    ax = ax or new_ax()\n",
+    "    ax.set_title('Rewards')\n",
+    "    colors = ['green', 'red']\n",
+    "    styles = ['-', '-.']\n",
+    "    df = cb_df(examples)\n",
+    "    for i, p in enumerate(people):\n",
+    "        for j, t in enumerate(topics):\n",
+    "            d = df[['reward']][(df['person']==p) & (df['chosen']==j)].rolling(window=window).mean()\n",
+    "            sns.lineplot(x = d.index, y=d['reward'],\n",
+    "                         label=f'E(r|{p},{t})', color = colors[j], linestyle = styles[i], ax=ax,\n",
+    "                         ci=None, sort=False, estimator=None)       \n",
+    "    ax.legend()\n",
+    "\n",
+    "    \n",
+    "def plot_probs(examples, job, ax=None, window=100):\n",
+    "    ax = ax or new_ax()\n",
+    "    ax.set_title('P(action|person)')\n",
+    "    ax.set_ylim(-0.05,1.05)\n",
+    "    colors = ['green', 'red']\n",
+    "    styles = ['-', '-.']\n",
+    "    df = pd.concat([cb_df(examples), job[0].predictions('-p').cb], axis=1)\n",
+    "    for i, p in enumerate(people):\n",
+    "        for j, t in enumerate(topics):\n",
+    "            d = df[(df['person']==p)][[str(j)]].rolling(window=window).mean() + i * 0.02\n",
+    "            sns.lineplot(x = d.index, y=d[str(j)],\n",
+    "                         label=f'P({t}|{p})',color = colors[j], linestyle = styles[i], ax=ax,\n",
+    "                         ci=None, sort=False, estimator=None)            \n",
+    "    ax.legend(loc='center left', bbox_to_anchor=(0.9, 0.5)) \n",
+    "     "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Interact"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "layout = [\n",
+    "    [plot_env, plot_probs]]\n",
+    "\n",
+    "dashboard = Dashboard(['-p'], layout, figsize=(12,4))\n",
+    "\n",
+    "playground = VwPlayground(my_cb_simulation, dashboard)\n",
+    "\n",
+    "playground.run(\n",
+    "    simulator_grid = {\n",
+    "        'n': IntSlider(value=10000, min=100, max=50000),\n",
+    "        'swap_after': IntSlider(value=5000, min=10, max=50000),\n",
+    "        'variance': FloatSlider(value=0, min=0, max=1, step=0.1),\n",
+    "        'bad_features': FloatLogSlider(value=1, min=0, max=10, base=2, step=1),\n",
+    "    },\n",
+    "    vw_grid = {\n",
+    "        '#base': fixed('--cb_explore_adf --dsjson --power_t 0 -P 100'),\n",
+    "        '--learning_rate': FloatLogSlider(value=2**(-20), min=-20, max=10, base=2, step=1),\n",
+    "        '--cb_type': fixed('mtr'),\n",
+    "        '--l1': FloatLogSlider(value=2**(-20), min=-20, max=10, base=2, step=1),\n",
+    "        '#interactions': ['', '-q sa', '-q ::'],\n",
```

Comment:
do we also want to try with `b` namespace or it is covered in `q ::`

Reply:
you can add it if you like, but it should be probably similar to '' (with slower learning, but it is harder to observe since regret of trivial policy is much smaller in this case)